### ISSUE TITLE
[Bug] DBT Dynamic Table Builds Failing when a model.yml file is added; succeeds otherwise. 

### ISSUE THREAD

**Issue by tan29cross** [dbt-labs/dbt-snowflake#977]
### Is this a new bug in dbt-snowflake?

- [X] I believe this is a new bug in dbt-snowflake
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I had built a dynamic table with a refresh frequency of 2 mins, it builds alright when I don't add a .yml file (test) to this model, but throws an unexpected error like this: 
![image](https://github.com/dbt-labs/dbt-snowflake/assets/29129015/f1b6677c-bdad-4115-8aba-dbe6cb3ff091)

### Expected Behavior

Ideally, the model build should succeed with or without the .yml file. 

### Steps To Reproduce

1. Create a dynamic table model 
2. Add .yml file for the model 
3. Run dbt-build --select <model name>

### Environment

```markdown
- OS: Mac OS 14.4.1 
- Python:3.11
- dbt-core:1.6.5
- dbt-snowflake:1.6.4
```

**Comment by tan29cross** [comment]
Worked after updating the snowflake adapter. You can go ahead and close this issue.

**Comment by dataders** [comment]
closing as resolved by upgrading to `dbt-snowflake 1.7`
